CoverMyDotNet.Tests/ClientTests.cs

```diff
@@ -204,6 +205,233 @@ public void Should_Client_Delete_Post_Correct_Data()
 				Assert.GreaterOrEqual(requestContent.IndexOf("Steve"), 0);
 			}	
 		}
-	}
-}
 
+		[Test]
+		public void Should_Client_Search_Drug_Use_Correct_Data()
+		{
+			string drug_name = Guid.NewGuid().ToString();
+			Dictionary<string, string> requestContent = new Dictionary<string, string>();
+			var requestHandlers = new List<MockHttpHandler>()
+			{
+				new MockHttpHandler("/drugs", "GET", (req, rsp, prm) => 
+				{
+					requestContent = prm;
+				})
+			};
+			using(new MockServer(DEFAULT_MOCK_PORT, requestHandlers))
+			{
+				_client.SearchDrugs(drug_name);
+				Assert.That(requestContent.Any(p => p.Key == "q"));
+				Assert.That(requestContent.Any(p => p.Value == drug_name));
+			}	
+		}
+
+		[Test]
+		public void Should_Client_Search_Drug_Parse_Data()
+		{
+			string drug_name = Guid.NewGuid().ToString();
+			Dictionary<string, string> requestContent = new Dictionary<string, string>();
+			var requestHandlers = new List<MockHttpHandler>()
+			{
+				new MockHttpHandler("/drugs", "GET", (req, rsp, prm) => 
+				{
+					return File.ReadAllText("Fixtures/Drugs.json");
+				})
+			};
+			using(new MockServer(DEFAULT_MOCK_PORT, requestHandlers, (req, rsp, prm) => rsp.Header("Content-Type", "application/json")))
+			{
+				var drugs = _client.SearchDrugs(drug_name);
+				Assert.AreEqual(drugs.Drugs.Count, 3);
+				Assert.AreEqual(drugs.Drugs.First().FullName, "Boniva 150MG tablets");
+				Assert.AreEqual(drugs.Drugs.First().Strength, "150");
+				Assert.AreEqual(drugs.Drugs[1].Href, "https://staging.api.covermymeds.com/drugs/094563");
+				Assert.AreEqual(drugs.Drugs[2].GPI, "30042048102030");
+			}	
+		}
+
+
+		[Test]
+		public void Should_Client_Get_Drug_Use_Correct_Data()
+		{
+			string drug_id = Guid.NewGuid().ToString();
+			bool called = false;
+			var requestHandlers = new List<MockHttpHandler>()
+			{
+				new MockHttpHandler(string.Format("/drugs/{0}", drug_id), "GET", (req, rsp, prm) => 
+				{
+					called = true;
+				})
+			};
+			using(new MockServer(DEFAULT_MOCK_PORT, requestHandlers))
+			{
+				_client.GetDrug(drug_id);
+				Assert.That(called);
```

Comment:
I don't know anything about MockHttpHandler, only conceptually what it is doing here, but I'd expect it to have some kind of method to assert that a request was made. (In lieu of using a local variable for this.)  Is there anything like that in the usage documentation?